CONTRIBUTING.md

```diff
@@ -40,13 +39,45 @@ tox                      # runs 'lint' and 'unit' environments
 
 ## Build charm
 
-Build the charm in this git repository using:
+The build environment assumes that there are preinstalled on the system:
+* [tox](https://tox.wiki/) (version 4+ !!!)
+* [poetry](https://python-poetry.org/)
+* [charmcraft](https://snapcraft.io/charmcraft)
+* [charmcraftcache](https://pypi.org/project/charmcraftcache/)
+* [pipx](https://python.land/virtual-environments/pipx)
+* [libpq-dev](https://www.postgresql.org/docs/current/libpq.html)
+
+To build the charm it is also necessary at least 5GB if free disk space and
+it is recommended to provide 4+ CPU cores and 8GB+ RAM for a decent build speed.
+
+To install all above build dependencies (assuming you are on Ubuntu 22.04 LTS):
 
 ```shell
-tox run -e build-dev
+sudo snap install charmcraft --classic
+
+sudo snap install lxd # should be pre-installed on 22.04
+lxd init --auto       # init LXD (if never used earlier)
+
+sudo apt update && sudo apt install --yes libpq-dev pipx
```

Comment:
Yes, pipx manuals use apt for 23.04+ only.
In the same time, for 22.04, they recommend to install pipx via pip. Which is missing on bare 22.04 => we should install pip first (from apt). Installing pip to install pipx is an overkill, IMO.

In the same time, we have tested pipx from apt on 22.04 and it does the job well.

Do you mind to keep this AS IS?
Otherwise users can choose different installations method and troubleshoot pipx installation instead of hacking the charm. :-)

Reply:
Yes, once we add 24.04 support for our charms => 24.10 roadmap.
Keeping this AS IS of no strong objection.